### PR TITLE
Add loopback action test cases

### PR DESCRIPTION
**Pull request by nhe-NV**
### Description of PR

Summary: Add test cases for loopback action
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012

#### What is the motivation for this PR?
Add new test cases for loopback action
#### How did you do it?
Add 3 testcases for the loopback action feature:
test_loopback_action_basic
test_loopback_action_port_flap
test_loopback_action_reload
#### How did you verify/test it?
Run all the new test cases, and tests pass
#### Any platform specific information?
No
#### Supported testbed topology if it's a new test case?
any
### Documentation
https://github.com/sonic-net/SONiC/blob/master/doc/ip-interface/loopback-action/rif_loopback_action_testplan.md
https://github.com/sonic-net/SONiC/blob/master/doc/ip-interface/loopback-action/ip-interface-loopback-action-design.md

### CODE REVIEW

**Comment by liat-grozovik** [comment]
@nhe-NV could you please handle LGTM new alerts and build failures?

File: tests/common/plugins/allure_wrapper/allure_step_wrapper.py
Comment (roysr-nv) [inline tests/common/plugins/allure_wrapper/allure_step_wrapper.py:1]: well done, we should start using it for any new test, please inform the team

File: tests/iface_loopback_action/conftest.py
Comment (roysr-nv) [inline tests/iface_loopback_action/conftest.py:190]: remove_ori_dut_port_config

orig sounds more natural to me than ori

File: tests/iface_loopback_action/conftest.py
Comment (roysr-nv) [inline tests/iface_loopback_action/conftest.py:196]: why wasn't it merged with 'setup' fixture?

File: tests/iface_loopback_action/test_iface_loopback_action.py
Comment (roysr-nv) [inline tests/iface_loopback_action/test_iface_loopback_action.py:32]: action

**Review by roysr-nv — COMMENTED** [review]
All in all looking good, Nana. Please review the minor comments. What is the test runtime?

**Comment by nhe-NV** [comment]
> This pull request **introduces 1 alert** when merging [192c2e9](https://github.com/Azure/sonic-mgmt/commit/192c2e93744e596409747f94365e89d5f31a8d0e) into [7ae74e4](https://github.com/Azure/sonic-mgmt/commit/7ae74e4a857e141fd3b5d14b9f53b65e8800b170) - [view on LGTM.com](https://lgtm.com/projects/g/Azure/sonic-mgmt/rev/pr-bd41bf081d4f84d4c9da842cee79b2f449188fe1)
> 
> **new alerts:**
> 
> * 1 for Unused import

Hi, I have added "# lgtm[py/unused-import]" for the fixture which is used in the conftest.py, but is detected as "Unused import" , but it still pop up this alerts.

File: tests/common/plugins/allure_wrapper/allure_step_wrapper.py
Comment (nhe-NV) [reply tests/common/plugins/allure_wrapper/allure_step_wrapper.py:1]: > All in all looking good, Nana. Please review the minor comments. What is the test runtime?

It takes less than 30 mins to finish all the test cases

File: tests/iface_loopback_action/test_iface_loopback_action.py
Comment (nhe-NV) [reply tests/iface_loopback_action/test_iface_loopback_action.py:32]: Fixed

File: tests/iface_loopback_action/conftest.py
Comment (nhe-NV) [reply tests/iface_loopback_action/conftest.py:190]: > remove_ori_dut_port_config
> 
> orig sounds more natural to me than ori

Fixed

**Comment by nhe-NV** [comment]
> 

It takes less than 30 mins to finish all the test cases

**Comment by nhe-NV** [comment]
> 



> All in all looking good, Nana. Please review the minor comments. What is the test runtime?

It takes less than 30 mins to finish all the test cases

File: tests/iface_loopback_action/conftest.py
Comment (nhe-NV) [reply tests/iface_loopback_action/conftest.py:196]: > 

it can not be merged, if merged together, then teardown for the fixture backup_and_restore_config_db_package will be executed late then the recover_config, it is not expected.

**Comment by liat-grozovik** [comment]
@roysr-nv could you please help to review following the changes requested?

**Comment by liat-grozovik** [comment]
@roysr-nv please review

**Review by roysr-nv — APPROVED** [review]

File: tests/iface_loopback_action/iface_loopback_action_helper.py
Comment (liorghub) [inline tests/iface_loopback_action/iface_loopback_action_helper.py:144]: Consider add return None to make it more clear.

File: tests/iface_loopback_action/iface_loopback_action_helper.py
Comment (liorghub) [inline tests/iface_loopback_action/iface_loopback_action_helper.py:134]: Typo Noe

File: tests/iface_loopback_action/iface_loopback_action_helper.py
Comment (nhe-NV) [reply tests/iface_loopback_action/iface_loopback_action_helper.py:309]: I did not find such module

**Review by roysr-nv — DISMISSED** [review]

File: tests/iface_loopback_action/test_iface_loopback_action.py
Comment (wangxin) [inline tests/iface_loopback_action/test_iface_loopback_action.py:28]: Will this still work if allure is not configured?

Comment (nhe-NV) [reply tests/iface_loopback_action/test_iface_loopback_action.py:28]: Yes, it works, and it is not the first time we use the allure in sonic-mgmt, such as in tests/ecmp/inner_hashing/test_inner_hashing_lag.py, we already used the allure.

**Comment by zhangyanzhao** [comment]
waiting for build success. @wangxin  @roysr-nv @liorghub can you please approve this PR if the review is ok? We need merge this PR to get the feature into 202211 release. Thanks.

**Review by roysr-nv — DISMISSED** [review]

**Review by wangxin — APPROVED** [review]

**Pull request merged**
